app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationScreen.kt

```diff
@@ -309,40 +296,40 @@ private fun ConversationScreen(
     onJoinCall: () -> Unit,
     onReactionClick: (messageId: String, reactionEmoji: String) -> Unit,
     onResetSessionClick: (senderUserId: UserId, clientId: String?) -> Unit,
-    onMentionMember: (String?) -> Unit,
     onUpdateConversationReadDate: (String) -> Unit,
     onDropDownClick: () -> Unit,
     onBackButtonClick: () -> Unit,
     composerMessages: SharedFlow<SnackBarMessage>,
     conversationMessages: SharedFlow<SnackBarMessage>,
     conversationMessagesViewModel: ConversationMessagesViewModel,
-    onPingClicked: () -> Unit,
     onSelfDeletingMessageRead: (UIMessage.Regular) -> Unit,
-    currentSelfDeletionTimer: SelfDeletionTimer,
     onNewSelfDeletingMessagesStatus: (SelfDeletionTimer) -> Unit,
     tempWritableImageUri: Uri?,
     tempWritableVideoUri: Uri?,
-    onFailedMessageRetryClicked: (String) -> Unit
+    onFailedMessageRetryClicked: (String) -> Unit,
+    requestMentions: (String) -> Unit,
+    onClearMentionSearchResult: () -> Unit
 ) {
-    val conversationScreenState = rememberConversationScreenState()
-    val messageComposerState = rememberMessageComposerState()
     val context = LocalContext.current
+    val conversationScreenState = rememberConversationScreenState()
+    val messageComposerState = rememberMessageComposerStateHolder(
+        messageComposerViewState = messageComposerViewState,
+        modalBottomSheetState = conversationScreenState.modalBottomSheetState,
+        tempWritableImageUri = tempWritableImageUri,
+        tempWritableVideoUri = tempWritableVideoUri,
+    )
 
     LaunchedEffect(conversationMessagesViewModel.savedStateHandle) {
         // We need to check if we come from the media gallery screen and the user triggered any action there like reply
         conversationMessagesViewModel.checkPendingActions(
             onMessageReply = {
                 withSmoothScreenLoad {
-                    messageComposerState.reply(it)
+                    messageComposerState.toReply(it)
```

Comment:
It seems we're still having the UI / Composable decide stuff, instead of leaving them to the ViewModel.

We are unable to test this kind of things at the moment.

Reply:
ConversationScreen needs a refactor